src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,369 @@ title: Builder.io & Astro
 
-## Community Resources
+* **A Builder account and space** - If you don't have an account yet, [sign up for free](https://www.builder.io/) and create a new space. If you already have a space with Builder, feel free to use it, but you will need to modify the code to match the model name (`blogpost`) and custom data fields.
+* **A Builder API key** - This public key will be used to fetch your content from Builder. [Read Builder's guide on how to find your key](https://www.builder.io/c/docs/using-your-api-key#finding-your-public-api-key).
+
+## Setting up credentials
+
+To add your Builder API key and your Builder model name to Astro, create a `.env` file in the root of your project (if one does not already exist) and add the following variables:
+
+```ini title=".env"
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+BUILDER_BLOGPOST_MODEL='blogpost'
+```
+
+Now, you should be able to use this API key in your project.
+
+:::note
+At the time of writing, [this key is public](https://www.builder.io/c/docs/using-your-api-key#prerequisites), so you don't have to worry about hiding or encrypting it.
+:::
+
+If you would like to have IntelliSense for your environment variables, you can create a `env.d.ts` file in the `src/` directory and configure `ImportMetaEnv` like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly BUILDER_API_PUBLIC_KEY: string;
+}
+```
+
+Your project should now include these files:
+
+<FileTree title="Project Structure">
+- src/
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+
+## Making a blog with Astro and Builder
+
+### Creating a model for a blog post
+
+The instructions below create an Astro blog using a Builder model (Type: "Section") called `blogpost` that contains two required text fields: `title` and `slug`.
+
+:::tip[For visual types]
+You can find videos showing this procedure in one of Builder's official tutorials [here](https://www.builder.io/blog/creating-blog#creating-a-blog-article-model).
+:::
+
+In the Builder app create the model that will represent a blog post: go to the **Models** tab and click the **+ Create Model** button to create model with the following fields and values:
+
+- **Type:** Section
+- **Name:** "blogpost"
+- **Description:** "This model is for a blog post"
+
+In your new model use the **+ New Custom Field** button to create 2 new fields:
+
+1. Text field
+    - **Name:** "title"
+    - **Required:** Yes
+    - **Default value** "I forgot to give this a title"
+
+    (leave the other parameters as their defaults)
+
+2. Text field
+    - **Name:** "slug"
+    - **Required:** Yes
+    - **Default value** "some-slugs-take-their-time"
+
+    (leave the other parameters as their defaults)
+
+Then click the **Save** button in the upper right. 
+
+:::caution
+This guide relies on these fields being exactly like this and later code will not run correctly if these fields don't exist, are empty, or the titles misspelled.
+:::
```

Comment:
```suggestion
```

Suggest remove this first caution.

As in my other comment, two notes/asides and ESPECIALLY two cautions next to each other are really bad practice. I strongly feel this one is unneeded. It's basically saying: Here are instructions. WARNING YOU HAVE TO FOLLOW THEM. (That's what instructions are for? You follow them?)

Something like adding to around line 68 with "creating the REQUIRED fields exactly as specified below", i.e. working this idea inline that fields are not forgiving, can't be blank etc. is a better strategy. 

It targets the reader at the moment they are performing that step, rather than at the end a huge caution that will make people go, "Oh crap, maybe I should go back and check" and definitely breaks flow when it won't matter for most: they will have followed the instructions you gave them.

I know all instructions are subject to user error, but I prefer not to make people second guess themselves or worry unnecessarily, nor clutter the flow if we can help it. People who make a mistake will simply have to go back and recheck, and everyone else can move along.